src/Components/Web/src/WebRenderer.cs

```diff
@@ -145,3 +153,9 @@ public void RemoveRootComponent(int componentId)
             => _jsComponentInterop.RemoveRootComponent(componentId);
     }
 }
+
+[JsonSerializable(typeof(object[]))]
+[JsonSerializable(typeof(int))]
+[JsonSerializable(typeof(Dictionary<string, JSComponentConfigurationStore.JSComponentParameter[]>))]
+[JsonSerializable(typeof(Dictionary<string, List<string>>))]
+internal sealed partial class WebRendererSerializerContext : JsonSerializerContext;
```

Comment:
I'm curious about how you think we should maintain this in the future. How would a developer know when/what to add to this list of types? Am I right to think that if they start serializing something different and fail to add it here, everything would still work but would just be slower? Or would some error occur so they know to change this code?

Reply:
Also, how did you even knew to include these specific types? I'm hoping it's because if you don't, there's a clear error indicating they need to be included!

Reply:
> How would a developer know when/what to add to this list of types?

These types match the types of the arguments passed to the JS interop call. Failing to do this results in a runtime error. However, I've just pushed a change that makes it even clearer where these types come from, so hopefully that eliminates any confusion. If we end up reverting that change, I can add a comment explaining how these types should be specified.